internal/codespaces/api/api.go

```diff
@@ -60,10 +67,24 @@ type httpClient interface {
 
 // New creates a new API client with the given token and HTTP client.
 func New(token string, httpClient httpClient) *API {
+	serverURL := os.Getenv("GITHUB_SERVER_URL")
+	if serverURL == "" {
+		serverURL = githubServer
+	}
+	apiURL := os.Getenv("GITHUB_API_URL")
+	if apiURL == "" {
+		apiURL = githubAPI
+	}
+	vscsURL := os.Getenv("VSCS_TARGET_URL")
```

Comment:
I like that all this is now configurable, but this introduces 3 new environment variables:
- GITHUB_SERVER_URL
- GITHUB_API_URL
- VSCS_TARGET_URL

We already supported setting `GH_HOST` to switch between non-github.com hosts, so adding all of the above feels a bit redundant. Of course, now there is also the `https://online.visualstudio.com/api` endpoint in addition to the GitHub API, so that needs to be configurable, but I wonder should we still stick with a `GH_` prefix for anything that affects the GitHub CLI and name this variable something like `GH_VSCS_URL`?

Reply:
@mislav those env var names are important because they are being created as part of the Codespaces environment and are expected to be automatically picked up by the CLI. Note also that the Codespaces Typescript Extension also picks them up and therefore any changes to the environment variable keys would require changes to both the extension and the CLI _and_ our github/github repository which sends those env vars in the first place.